src/hooks/useKeyboardNavigation.js

```diff
@@ -27,11 +29,68 @@ const useKeyboardNavigation = ({
 
   useEffect(() => {
     return tinykeys(emojiListRef.current, {
-      ArrowRight: focusNextEmoji,
-      ArrowLeft: focusPrevEmoji,
+      ArrowRight: () => navigateGrid(RIGHT),
+      ArrowLeft: () => navigateGrid(LEFT),
+      ArrowUp: () => navigateGrid(UP),
+      ArrowDown: () => navigateGrid(DOWN),
     });
   }, []);
 
+  const updateActiveItem = newActiveItem => {
+    if (newActiveItem && newActiveItem.firstChild)
+      newActiveItem.firstChild.focus();
+  };
+
+  const navigateGrid = direction => {
```

Comment:
Since we're not using the refs in this function (or the others actually), everything that's not a useEffect can be outside of the hook. It will make the hook itself easier to read and reason about, and also help us see which part is actually interacting with react and which part isn't.